cob_cartesian_controller/include/cob_cartesian_controller/cartesian_controller_data_types.h

```diff
@@ -64,9 +68,140 @@ namespace cob_cartesian_controller
         unsigned int move_type;
         MoveLinStruct move_lin;
         MoveCircStruct move_circ;
+        ProfileStruct profile;
     };
 
-}//namespace
 
 
+    class PathArray{
+        public:
+            PathArray(double idx, double Se, double start_value, std::vector<double> array):
+                Se_(Se),
+                array_(array),
+                start_value_(start_value)
+            {
+                idx_= idx;
+                calcTe_=false;
+            }
+
+            ~PathArray()
+            {
+                array_.clear();
+            }
+
+            unsigned int getIdx();
+            bool getCalcTe();
+            double getSe();
+            std::vector<double> getArray();
+            void setCalcTe(bool calcTe);
+            void setIdx(unsigned int idx);
+            void setArray(std::vector<double> array);
+            double getStartValue();
+
+        private:
+            unsigned int idx_;
+            bool calcTe_;
+            double Se_;
```

Comment:
I guess we just could make these members public and remove the respective get/set functions...